examples/psbt.rs

```diff
@@ -0,0 +1,615 @@
+    fn new(account_0_xpub: ExtendedPubKey, input_xpub: ExtendedPubKey, master_fingerprint: Fingerprint) -> Self {
+        WatchOnly { account_0_xpub, input_xpub, master_fingerprint }
+    }
+
+    /// Creates a PSBT.
+    fn create_psbt<C: Verification>(&self, secp: &Secp256k1<C>) -> Result<Psbt> {
+        let input = Txid::from_hex(INPUT_UTXO_TXID)?;
+        let vout = INPUT_UTXO_VOUT;
+
+        let to_address = Address::from_str(RECEIVE_ADDRESS)?;
+        let to_amount = Amount::from_str_in(OUTPUT_AMOUNT_BTC, Denomination::Bitcoin)?;
+
+        let (_, change_address, _) = self.change_address(secp)?;
+        let change_amount = Amount::from_str_in(CHANGE_AMOUNT_BTC, Denomination::Bitcoin)?;
+
+        let tx = Transaction {
+            version: 2,
+            lock_time: 0,
+            input: vec![
+                TxIn {
+                    previous_output: OutPoint {
+                        txid: input,
+                        vout,
+                    },
+                    script_sig: Script::new(),
+                    sequence: 0xFFFFFFFF, // Ignore nSequence.
+                    witness: Witness::default(),
+                },
+            ],
+            output: vec![
+                TxOut {
+                    value: to_amount.as_sat(),
+                    script_pubkey: to_address.script_pubkey(),
+                },
+                TxOut {
+                    value: change_amount.as_sat(),
+                    script_pubkey: change_address.script_pubkey(),
+                }
+            ],
+        };
+
+        let psbt = Psbt::from_unsigned_tx(tx)?;
+
+        Ok(psbt)
+    }
+
+    /// Updates the PSBT.
+    ///
+    /// This is the first update so inputs/outputs fields are guaranteed by the spec to be empty.
+    fn update_psbt<C: Verification>(&self, secp: &Secp256k1<C>, mut psbt: Psbt) -> Result<Psbt> {
+        let mut input = Input::default();
+
+        let witness_utxo = TxOut {
+            value: Amount::from_str_in(INPUT_UTXO_VALUE, Denomination::Bitcoin)?.as_sat(),
+            script_pubkey: Script::from_hex(INPUT_UTXO_SCRIPT_PUBKEY)?,
+        };
+        input.witness_utxo = Some(witness_utxo);
+
+        let pk = self.input_xpub.to_pub();
+        let wpkh = pk.wpubkey_hash().expect("a compressed pubkey");
+
+        let redeem_script = Script::new_v0_p2wpkh(&wpkh);
+        input.redeem_script = Some(redeem_script);
+
+        let witness_script = Script::new_p2pkh(&pk.pubkey_hash());
+        input.witness_script = Some(witness_script);
+
+        let fingerprint = self.master_fingerprint;
+        let path = input_derivation_path()?;
+        input.bip32_derivation = BTreeMap::from([(pk.inner, (fingerprint, path))]);
+
+        let ty = PsbtSighashType::from_str("SIGHASH_ALL").map_err(|_| Error::SighashTypeParse)?;
+        input.sighash_type = Some(ty);
+
+        psbt.inputs = vec![input];
+
+        let send = Output::default(); // FIXME: Do we need to update the send output?
```

Comment:
Though the updater has enough information to add `redeem_scripts` to `send` and `change` but there is no need. the unsigned_tx (PSBTv1 only) has the addresses which include the standard p2wpkh scripts already. With the input `redeem_script` set properly, the output of this program is a valid and correct bitcoin transaction as far as I can tell.